adapters/repos/db/vector/hnsw/neighbor_connections.go

```diff
@@ -188,6 +141,10 @@ func (n *neighborFinderConnector) connectNeighborAtLevel(neighborID uint64,
 
 	neighbor.Lock()
 	defer neighbor.Unlock()
+	if level > neighbor.level {
+		// upgrade neighbor level if the level is out of sync due to a delete re-assign
+		neighbor.upgradeToLevelNoLock(level)
+	}
```

Comment:
Also not tested (I'll stop now, but could you go through the changeset and add tests for the non-error pathways?

Reply:
For the other cases above (marshalling in between types, etc.) it should be easy to add tests, I can add those. For this one the reason I don't have any is in the PR description:

> This is the only one of the fixes that have no explicit test in Weaviate as the chaos pipeline clearly reproduced this issue with all the other issues fixed. It's not 100% ideal to only rely on the chaos pipeline because of the late feedback, but I couldn't think of a "good" way to write a test for this that doesn't run for ages, and also this PR that was supposed to be 50 lines, has grown substantially because of all the other bugs that were discovered and fixed.

Reply:
alright!